slam-libraries/viam-cartographer/src/mapping/map_builder.h

```diff
@@ -38,9 +38,18 @@ class MapBuilder {
 
     // SaveMapToFile saves the current map_builder_ state to a pbstream file at
     // the provided path.
-    void SaveMapToFile(bool include_unfinished_submaps,
+    bool SaveMapToFile(bool include_unfinished_submaps,
                        const std::string filename_with_timestamp);
 
+    // SaveMapToStream saves the current map_builder_ state to the pbstream
+    // stream provided via a temporary pbstream file.
+    std::string SaveMapToStream(const std::string filename_with_timestamp,
```

Comment:
[nit] I would move this to slam_service.h, since it doesn't access any of the data in the map builder.